### DISCUSSION TITLE
Can sanni cart reader reflash the repro wonderswan carts？（as well as dump save files）

### DISCUSSION THREAD

**Discussion by hualazimo7** [sanni/cartreader#948] [General Chat]
I have designed a homemade repro cart, which uses mx29lv640ebti as rom and Fm28V020 ---a Fram as ram.
I just want to know can I use the V2 WS adapter  to reflash the repro one and dump and write the save file.

The rom flash's address and data I/O have been connected with the edge fingers， and the diy repro cart works well.
 (flash was programmed before assembled)

The ROM flash could be other 29 flash

**Comment by hualazimo7** [comment]
![Screenshot_2024-05-24-13-24-02-818_com.tencent.mobileqq-edit.jpg](https://github.com/sanni/cartreader/assets/74229525/cdf01cf1-7cf5-47e7-abbe-9ca9e360c640)

**Reply by geoffreyplatt** [reply]
This looks almost identical to the luilili_420 design.. I've been looking for something like this myself to help a friend publish his game. Is your design going to be opensource? Regarding reflashing with the Sanni, i am not sure. I believe it would be possible as other repro carts can be reflashed.. You might also want to look into the reflashers from Game bank (Tengu)

**Reply by hualazimo7** [reply]
A homebrew game? Sorry for replying so late, actually my friend has designed a prototype of flasher, this cartridge works well at now.
You can just use the stock cart which with sop44 maskrom flash. Just replace that flash with a refalshable one and lift it's  "we" pin, wire it to the cartridge 46pin also "we" pin. So you can write your own games.

BTW, can you share your friend's game flile?

**Reply by geoffreyplatt** [reply]
Unfortunately, I cannot share the game file as he has not released the game yet. I am working on helping him publish it and have been looking into options. I own a reproduction and mod company called Guinea pig games in Florida.

**Reply by geoffreyplatt** [reply]
Do you have any plans on open sourcing your cartridge design?

**Comment by StanMOI** [comment]
> I have designed a homemade repro cart, which uses mx29lv640ebti as rom and Fm28V020 ---a Fram as ram. I just want to know can I use the V2 WS adapter to reflash the repro one and dump and write the save file.
> 
> The rom flash's address and data I/O have been connected with the edge fingers， and the diy repro cart works well. (flash was programmed before assembled)
> 
> The ROM flash could be other 29 flash

PCB with FRAM ?
FRAM => Synchronous SRAM = Happens at the same time. Requires a clock signal to validate its control signals.
SRAM => Asynchronous SRAM = Doesn't happen at the same time. Is accessed without a clock.

If you use FRAM you need some NOR GATE chip to make it works with SNES.
FRAM need "clock signal" , Without that "Clock signal" SNES can not access to FeRAM.
